actix-web-codegen/src/lib.rs

```diff
@@ -155,7 +63,52 @@ pub fn patch(args: TokenStream, input: TokenStream) -> TokenStream {
 /// - `wrap="Middleware"` - Registers a resource middleware.
 #[proc_macro_attribute]
 pub fn route(args: TokenStream, input: TokenStream) -> TokenStream {
-    route::generate(args, input, route::GuardType::Multi)
+    route::with_method(None, args, input)
+}
+
+macro_rules! doc_comment {
+    ($($x:expr)*; $($tt:tt)*) => {
+        $(#[doc = $x])*
+        $($tt)*
+    };
+}
+
+macro_rules! method_macro {
+    (
+        $(
+            ($method:ident, $variant:ident, $upper:ident);
+        )+
+    ) => {
+        $(
+            doc_comment! {
+                concat!("Creates route handler with `", stringify!($upper), "` method guard.")
+                concat!("")
+                concat!("Syntax: `#[", stringify!($method), "(\"path\" [, attributes])]`")
+                concat!("")
+                concat!("## Attributes:")
+                concat!("")
+                concat!("- `\"path\"` - *required* Raw literal string with path for which to register handler")
```

Comment:
i don't think this required marker is necessary, it's specified in the syntax part